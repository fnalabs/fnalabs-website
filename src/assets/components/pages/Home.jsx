import React, { Component } from 'react'
import Helmet from 'react-helmet'
import { Cookies } from 'react-cookie-consent'
import ReactGA from 'react-ga'

import { AnalyticsLink, Container, Tile } from 'common'
import { Hero, HeroBody } from 'common/hero'
import { HiveIO } from 'icons'

import meta from 'metadata'

export default class Home extends Component {
  componentDidMount () {
    if (Cookies.get('CookieConsent')) {
      ReactGA.pageview(this.props.location.pathname, undefined, meta.common.siteName)
    }
  }

  render () {
    const { description, url } = meta['/']
    const siteName = meta.common.siteName

    return (
      <>
        <Helmet>
          <title>{siteName}</title>
          <meta name='description' content={description} />
          <link rel='canonical' href={url} />

          <meta property='og:title' content={siteName} />
          <meta property='og:description' content={description} />
          <meta property='og:site_name' content={siteName} />
          <meta property='og:url' content={url} />
          <meta property='og:type' content='website' />
        </Helmet>

        <Hero size='fullheight' color='light' bold>
          <HeroBody>
            <Container>
              <Tile type='ancestor'>
                <Tile type='parent'>
                  <Tile type='child' size={6} vertical centered>
                    <h1 className='title'>
                      <HiveIO className='svg-inline' height='72' />
                      Hive<sup>io</sup>
                    </h1>
                    <h2 className='subtitle'>A reactive, cloud-native framework for building RESTful microservices.</h2>
                    <AnalyticsLink to='https://hiveframework.io' className='button is-link' target='_blank' rel='noopener noreferrer'>Learn More</AnalyticsLink>
                  </Tile>

                  <Tile type='child' size={6}>
                    <div className='svg-animation-container'>
                      <div className='svg-animation is-mongo-svg is-svg-1' />
                      <div className='svg-animation is-hive-svg is-svg-2' />
                      <div className='svg-animation is-envoy-svg is-svg-3' />

                      <div className='svg-animation is-elastic-svg is-svg-4' />
                      <div className='svg-animation is-hive-svg is-svg-5' />
                      <div className='svg-animation is-hive-svg is-svg-6' />
                      <div className='svg-animation is-envoy-svg is-svg-7' />

                      <div className='svg-animation is-redis-svg is-svg-8' />
                      <div className='svg-animation is-kafka-svg is-svg-9' />
                      <div className='svg-animation is-hive-svg is-svg-10' />
                      <div className='svg-animation is-hive-svg is-svg-11' />
                      <div className='svg-animation is-hive-svg is-svg-12' />
                      <div className='svg-animation is-envoy-svg is-svg-13' />
                      <div className='svg-animation is-envoy-svg is-svg-14' />
                    </div>
                  </Tile>
                </Tile>
              </Tile>
            </Container>
          </HeroBody>
        </Hero>
      </>
    )
  }
}
