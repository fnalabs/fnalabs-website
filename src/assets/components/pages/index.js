// site's main pages
export { default as Home } from './Home'
export { default as Projects } from './projects/Projects'
export { default as About } from './about/About'
export { default as Support } from './support/Support'

// policy pages
export { Cookie } from './policies/Cookie'
export { Privacy } from './policies/Privacy'
