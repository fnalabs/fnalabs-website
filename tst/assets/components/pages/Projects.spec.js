/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import ReactGA from 'react-ga'

import { Projects } from 'pages'

jest.mock('react-ga')

describe('<Projects />', () => {
  it('should render Projects page static content', () => {
    const tree = renderer.create(<Projects location={{ pathname: '/projects' }} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  it('should render Projects page static content and report page view', () => {
    shallow(<Projects location={{ pathname: '/projects' }} />, {
      context: { isConsent: true }
    })

    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
