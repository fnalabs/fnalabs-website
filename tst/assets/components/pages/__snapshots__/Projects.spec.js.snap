// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render Projects page static content 1`] = `
Array [
  <section
    className="hero is-light is-bold has-nav-spacing"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <div
          className="columns"
        >
          <div
            className="column is-8-desktop is-offset-2-desktop"
          >
            <div
              className="tile is-ancestor"
            >
              <div
                className="tile is-parent is-3"
              >
                <div
                  className="tile is-child has-text-centered"
                >
                  <svg
                    style={
                      Object {
                        "marginTop": 12,
                      }
                    }
                    viewBox="0 0 117.31758 135.46667"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m 58.658792,6.7627869 52.802038,30.4852731 0,60.970543 L 58.658789,128.70387 5.8567507,98.2186 l 1.7e-6,-60.970543 z"
                      fill="none"
                      stroke="#daa520"
                      strokeWidth="12"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="tile is-parent is-vertical"
              >
                <div
                  className="tile is-child"
                >
                  <article
                    className="content"
                  >
                    <h1
                      className="title"
                    >
                      Hive
                      <sup>
                        io
                      </sup>
                    </h1>
                    <h2
                      className="subtitle is-5"
                    >
                      From startup to enterprise...
                    </h2>
                    <p>
                      FnA Labs has evolved enterprise architecture by giving you the Hive
                      <sup>
                        io
                      </sup>
                       framework. Our product works for fresh startups and large-scale corporations alike. It is 
                      <strong>
                        reactive
                      </strong>
                       and 
                      <strong>
                        cloud-native
                      </strong>
                      , giving you more time to spend solving your business challenges and less time on your infrastructure. Furthermore the framework is flexible in allowing you to use multiple patterns and architectures, as well as providing robustness with support of the JSON Schema and Flux Standard Action specifications.
                    </p>
                  </article>
                </div>
                <div
                  className="tile is-child"
                >
                  <a
                    className="button is-link"
                    href="https://hiveframework.io"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Release Candidate 3
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-10-desktop is-offset-1-desktop"
        >
          <h2
            className="title"
          >
            More Projects
          </h2>
          <div
            className="tile is-ancestor is-vertical"
          >
            <div
              className="tile is-parent is-paddingless"
            >
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child is-vertical box"
                >
                  <div
                    className="tile is-child"
                  >
                    <article
                      className="content"
                    >
                      <h3>
                        Model(JSON)
                        <sup>
                          js
                        </sup>
                      </h3>
                      <p>
                        Our Model JSON Serialization project is a Model class designed to serialize and transfer structured data within the services/apps you build or across networks. It can represent data Models as either immutable or not and uses schema validation with Schema(JSON)
                        <sup>
                          js
                        </sup>
                        .
                      </p>
                    </article>
                  </div>
                  <div
                    className="tile is-child"
                  >
                    <a
                      className="button is-link"
                      href="https://fnalabs.github.io/model-json-js/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child is-vertical box"
                >
                  <div
                    className="tile is-child"
                  >
                    <article
                      className="content"
                    >
                      <h3>
                        Schema(JSON)
                        <sup>
                          js
                        </sup>
                      </h3>
                      <p>
                        Our JSON Schema project is a Schema class designed to validate structured data based on the JSON Schema specification. It represents the schemas as immutable objects with the flexibility of making partial schema validations via Closures and Thunks.
                      </p>
                    </article>
                  </div>
                  <div
                    className="tile is-child"
                  >
                    <a
                      className="button is-link"
                      href="https://fnalabs.github.io/schema-json-js/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="tile is-parent is-paddingless"
            >
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child is-vertical box"
                >
                  <div
                    className="tile is-child"
                  >
                    <article
                      className="content"
                    >
                      <h3>
                        test-automation
                      </h3>
                      <p>
                        We created a reusable automated browser test framework atop the popular Angular Protractor. If used simultaneously with our 
                        <a
                          href="https://github.com/fnalabs/test-automation-starter"
                        >
                          starter kit
                        </a>
                        , it will allow you to run automated browser tests inside a controlled environment. With the use of Docker you are able to automatically have tests run when a file is saved for development or single run containers for CI/CD deployments.
                      </p>
                    </article>
                  </div>
                  <div
                    className="tile is-child"
                  >
                    <a
                      className="button is-link"
                      href="https://fnalabs.github.io/test-automation/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="tile is-parent"
              >
                <div
                  className="tile is-child is-vertical box"
                >
                  <div
                    className="tile is-child"
                  >
                    <article
                      className="content"
                    >
                      <h3>
                        Starter Kits
                      </h3>
                      <p>
                        Over the course of our open source development journey, we've identified common underlying structures in the projects we've built. We decided to abstract those structures as starter kits:
                      </p>
                      <ul>
                        <li>
                          <a
                            href="https://github.com/fnalabs/starter-npm-js"
                          >
                            NPM package
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://github.com/fnalabs/starter-react-js"
                          >
                            React SSR PWA
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://github.com/fnalabs/starter-koa2-js"
                          >
                            Koa2 service
                          </a>
                        </li>
                      </ul>
                    </article>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
